src/source/load_vector_tile.ts

```diff
@@ -38,46 +62,143 @@ export class DedupedRequest {
         this.scheduler = scheduler;
     }
 
-    request(key: string, metadata: any, request: any, callback: LoadVectorDataCallback): () => void {
-        const entry = this.entries[key] = this.entries[key] || {callbacks: []};
+    addToSchedulerOrCallDirectly({
+        callback,
+        metadata,
+        err,
+        result,
+    }: {
+        callback: LoadVectorDataCallback;
+        metadata: any;
+        err: Error | null | undefined;
+        result: any;
+    }) {
+        if (this.scheduler) {
+            this.scheduler.add(() => {
+                callback(err, result);
+            }, metadata);
+        } else {
+            callback(err, result);
+        }
+    }
+
+    getEntry = (key: string) => {
+        return (
+            this.entries[key] || {
+                // use a set to avoid duplicate callbacks being added when calling from queue
+                callbacks: new Set(),
+            }
+        );
+    };
+
+    request({key, metadata, requestFunc, callback, fromQueue}: DedupedRequestInput): Cancelable {
+        const entry = (this.entries[key] = this.getEntry(key));
+
+        const removeCallbackFromEntry = ({key, requestCallback}) => {
+            const entry = this.getEntry(key);
+            if (entry.result) return;
+            entry.callbacks.delete(requestCallback);
+            if (entry.callbacks.size) {
+                return;
+            }
+            if (entry.cancel) {
+                entry.cancel();
+            }
+            filterQueue(key);
+            delete this.entries[key];
+        };
+
+        let advanced = false;
+        const advanceImageRequestQueue = () => {
```

Comment:
`advanceImageRequestQueue` is when we send off our queued requests. 

Queued requests will be sent only when `numRequests` is less than our limit (and there are things to grab from the queue).
The loop will blast through queued requests until either the queue runs out, or `numRequests` gets too big again